include/pagmo_plugins_nonfree/worhp.hpp

```diff
@@ -0,0 +1,1203 @@
+                    libworhp,                                             // the library
+                    "WorhpSetBoolParam"                                   // name of the function to import
+                );
+            WorhpSetIntParam = boost::dll::import<bool(Params *, const char *, bool)>( // type of the function to import
+                libworhp,                                                              // the library
+                "WorhpSetIntParam"                                                     // name of the function to import
+            );
+            WorhpSetDoubleParam
+                = boost::dll::import<bool(Params *, const char *, bool)>( // type of the function to import
+                    libworhp,                                             // the library
+                    "WorhpSetDoubleParam"                                 // name of the function to import
+                );
+            WorhpFree = boost::dll::import<void(OptVar *, Workspace *, Params *,
+                                                Control *)>( // type of the function to import
+                libworhp,                                    // the library
+                "WorhpFree"                                  // name of the function to import
+            );
+            WorhpFidif = boost::dll::import<void(OptVar *, Workspace *, Params *,
+                                                 Control *)>( // type of the function to import
+                libworhp,                                     // the library
+                "WorhpFidif"                                  // name of the function to import
+            );
+        } catch (const std::exception &e) {
+            std::string message(
+                R"(
+An error occurred while loading the worhp library at run-time. This is typically caused by one of the following
+reasons:
+
+- The file declared to be the worhp library, i.e. )"
+                + m_worhp_library
+                + R"(, is not found or is found but it is not a shared library containing the necessary symbols 
+(is the file path really pointing to a valid shared library?)
+ - The library is found and it does contain the symbols, but it needs linking to some additional libraries that are not found
+at run-time.
+
+We report the exact text of the original exception thrown:
+
+ )" + std::string(e.what()));
+            pagmo_throw(std::invalid_argument, message);
+        }
+        // ------------------------- END WORHP PLUGIN -------------------------------------------------------------
+
+        m_log.clear();
+        auto fevals0 = prob.get_fevals();
+
+        // With reference to the worhp User Manual (V1.12)
+        // USI-0:  Call WorhpPreInit to properly initialise the (empty) data structures.
+        OptVar opt;
+        Workspace wsp;
+        Params par;
+        Control cnt;
+        WorhpPreInit(&opt, &wsp, &par, &cnt);
```

Comment:
We should not be calling this init function (and the corresponding ``WohrpFree()`` function at the end) manually. The correct way of doing it is with an RAII pattern, so if some exception is thrown during the evolution, the ``WorhpFree()`` function ends up being called before the exception is propagated. Like this, I assume we would be leaking memory or perhaps worse in case some error is thrown during the evolution.